pkg/collector/corechecks/cluster/orchestrator/processors/chunking.go

```diff
@@ -0,0 +1,125 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build orchestrator
+// +build orchestrator
+
+package processors
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/process/util"
+)
+
+// orchestratorMetadataList is a payload list of orchestratorMetadata payloads
+type orchestratorMetadataList struct {
+	orchestratorMetadataYaml [][]byte
+	orchestratorMetadata     []interface{}
+	chunker                  orchestratorMetadataChunker
+}
+
+// orchestratorMetadataChunker abstracts chunking of orchestratorMetadata payloads
+type orchestratorMetadataChunker interface {
+	// Accept takes a slice of orchestratorMetadata and allocates them to the current chunk
+	Accept(orchestratorMetadata []interface{}, weight int)
+}
+
+func (l *orchestratorMetadataList) Len() int {
+	return len(l.orchestratorMetadata)
+}
+
+func (l *orchestratorMetadataList) WeightAt(idx int) int {
+	if idx >= len(l.orchestratorMetadata) {
+		return 0
+	}
+	return len(l.orchestratorMetadataYaml[idx])
+}
+
+func (l *orchestratorMetadataList) ToChunk(start, end int, weight int) {
+	l.chunker.Accept(l.orchestratorMetadata[start:end], weight)
+}
+
+// chunkProps is used to track weight and size of chunks
+type chunkProps struct {
+	weight int
+	size   int
+}
+
+// chunkPropsTracker tracks weight and size of chunked payloads
+type chunkPropsTracker struct {
+	props []chunkProps
```

Comment:
I'm wondering whether we should move `chunkPropsTracker` and `chunkProps` to `pkg/process/util/chunking.go` as well because this part is quite common